app/ui/shared/widgets/list-item.jsx

```diff
@@ -22,7 +24,10 @@ const LIST_ITEM_STYLE = Style.registerStyle({
 
 class ListItem extends Component {
   shouldComponentUpdate(nextProps) {
-    return !isEqual(this.props, nextProps);
+    // Children in this component may contain immutable.js props, which are
```

Comment:
Updated the comment to 
```
    // Since this component uses plain js objects as props, we need to perform
    // deep quality checks on them to make sure we don't unnecessarily rerender.
    // However, there is no need to re-render this parent component when the
    // children change, which would happen when doing naive deep equality.
    // Furthermore, children in this component may contain immutable.js props,
    // which are throwing warnings when accessing some of their properties
    // while doing deep equality checks using lodash's `deepEqual`.
```